### ISSUE TITLE
Help please, I have problems with sync_upload

### ISSUE THREAD

**Issue by hectorzapata** [samwilson/phpflickr#6]
Hello, first of all I appreciate your support. I have been trying to upload an image using Laravel 5.6, I have generated the accessToken and accessTokenSecret with the permission 'delete' and when I try it returns me "The Flickr API returned the following error: # 99 - Insufficient permissions Method requires write privileges; none . "
I really do not know what I'm doing wrong, I also call the getRecent method and it works without problems which makes me think that the authentication is also correct.

I leave my code:

![image](https://user-images.githubusercontent.com/22927480/42343230-023087f0-805e-11e8-9ee8-ce920459f4c5.png)

Response:

![image](https://user-images.githubusercontent.com/22927480/42343841-fa6e25ca-805f-11e8-8324-9a0e542c3d07.png)

I should also mention that I changed the 436 line of the PHPFlickr.php file:
![image](https://user-images.githubusercontent.com/22927480/42343739-a30218dc-805f-11e8-98fc-ac10ce5565fe.png)

by:
![image](https://user-images.githubusercontent.com/22927480/42343766-b70350c6-805f-11e8-9a4c-5f2ae1539e8b.png)

Because I was returning the following error:
![image](https://user-images.githubusercontent.com/22927480/42343815-db938064-805f-11e8-9042-98fa35160216.png)

I really hope you can help me, thank you very much, regards

**Comment by lastcoolnameleft** [comment]
I'm experiencing the same issue, did you find a solution for it?

Thanks for suggesting the `new \CURLFile` line as I also hit that problem too with the signature.

**Comment by hectorzapata** [comment]
Yes, i found this tutorial http://www.developpeur-web.dantsu.com/fr/cours-tutoriels/flickr/a_authenticate-to-flickr-api-with-oauth-and-phpflickr, regards

**Comment by samwilson** [comment]
I'm very sorry about this bug. I'll try to get it sorted and the docs updated, this weekend. :)

**Comment by bart-d** [comment]
i'm down the same path. I had to replace the $args to new \CURLFile to get by the invalid signature. Now I have the same 'insufficient privileges' error while i went through get_auth_token.php to get write and even delete privileges. 

I also don't see how the other phpflickr/oauth instructions would help here. It's basically creating the same tokens.

**Comment by willpower232** [comment]
The problem is this function is lacking the oauth parameters completely so flickr is rejecting it by saying you have no permissions at all as there is no token in the request.

As flickr uploads to a completely different URL, I don't think there is a way within the current oauth library dependency to accomplish what we need but you can add a function into the package manually and update the phpflickr library yourself to use it.

In `vendor/lusitanian/oauth/src/OAuth/OAuth1/Service/Flickr.php` add this function to the class

```php
    public function getAuthorizationForPostingToAlternateURL($args, $urioverride)
    {
        $token = $this->storage->retrieveAccessToken($this->service());

        $this->signature->setTokenSecret($token->getAccessTokenSecret());
        $authParameters = $this->getBasicAuthorizationHeaderInfo();
        if (isset($authParameters['oauth_callback'])) {
            unset($authParameters['oauth_callback']);
        }

        $authParameters = array_merge($authParameters, array('oauth_token' => $token->getAccessToken()));

        $signatureParams = array_merge($authParameters, $args);

        $authParameters['oauth_signature'] = $this->signature->getSignature(new Uri($urioverride), $signatureParams);

        return array_merge($authParameters, $args);
    }
```

I borrowed those lines from `buildAuthorizationHeaderForAPIRequest` in `vendor/lusitanian/oauth/src/OAuth/OAuth1/Service/AbstractService.php`.

To update phpflickr, the lines around `realpath($photo)` need to look like this

```php
    $oauthService = $this->getOauthService();
    $args = $oauthService->getAuthorizationForPostingToAlternateURL($args, $this->upload_endpoint);

    // if (!empty($this->secret)) {
    //     $api_sig = md5($this->secret . $auth_sig);
    //     $args["api_sig"] = $api_sig;
    // }

    $photo = realpath($photo);
    $args['photo'] = new \Curlfile($photo);
```

I don't know the best way to proceed from here @samwilson, hopefully you have a cleverer way than I!

**Comment by sfatfarma** [comment]
The above worked for me also. Thank you very much!

**Comment by sfatfarma** [comment]
Thank you.
